tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -85,27 +86,67 @@ def test_overwrite_sg(region, custom_security_group, pcluster_config_reader, clu
         assert_that(mount_target_security_groups[0]).is_equal_to(custom_security_group_id)
         assert_that(mount_target_security_groups).is_length(1)
 
+    if scheduler == "slurm":
+        _check_connections_between_head_node_and_compute_nodes(cluster)
+        updated_config_file = pcluster_config_reader(
+            config_file="pcluster.config.update.yaml",
+            vpc_security_group_id=custom_security_group_id,
+        )
+        cluster.update(str(updated_config_file), force_update="true")
+        _check_connections_between_head_node_and_compute_nodes(cluster)
+
+
+def _check_connections_between_head_node_and_compute_nodes(cluster):
+    username = get_username_for_os(cluster.os)
+    head_node_public_ip = cluster.describe_cluster_instances(node_type="HeadNode")[0].get("publicIpAddress")
+    head_node_private_ip = cluster.describe_cluster_instances(node_type="HeadNode")[0].get("privateIpAddress")
+    head_node_username_ip = f"{username}@{head_node_public_ip}"
+    compute_nodes = cluster.describe_cluster_instances(node_type="Compute")
+    head_node_remote_command_executor = RemoteCommandExecutor(cluster)
+    for compute_node in compute_nodes:
+        compute_node_ip = compute_node.get("privateIpAddress")
+        head_node_remote_command_executor.run_remote_command(f"ping {compute_node_ip} -c 5")
+        compute_remote_command_executor = RemoteCommandExecutor(
+            cluster, compute_node_ip=compute_node_ip, bastion=head_node_username_ip
+        )
+        compute_remote_command_executor.run_remote_command(f"ping {head_node_private_ip} -c 5")
+        for other_compute_node in compute_nodes:
+            other_compute_node_ip = other_compute_node.get("privateIpAddress")
+            compute_remote_command_executor.run_remote_command(f"ping {other_compute_node_ip} -c 5")
+
 
 @pytest.fixture(scope="class")
 def custom_security_group(vpc_stack, region, request, cfn_stacks_factory):
     template = Template()
     template.set_version("2010-09-09")
     template.set_description("custom security group stack for testing additional_sg and vpc_security_group_id")
+    vpc_id = vpc_stack.cfn_outputs["VpcId"]
     security_group = template.add_resource(
         SecurityGroup(
             "SecurityGroupResource",
             GroupDescription="custom security group for testing additional_sg and vpc_security_group_id",
-            VpcId=vpc_stack.cfn_outputs["VpcId"],
+            VpcId=vpc_id,
         )
     )
+    vpc_cidr = boto3.client("ec2").describe_vpcs(VpcIds=[vpc_id])["Vpcs"][0]["CidrBlock"]
     template.add_resource(
         SecurityGroupIngress(
-            "SecurityGroupIngressResource",
+            "SecurityGroupSelfIngressResource",
             IpProtocol="-1",
             FromPort=0,
             ToPort=65535,
-            SourceSecurityGroupId=Ref(security_group),
+            CidrIp=vpc_cidr,
+            GroupId=Ref(security_group),
+        )
+    )
+    template.add_resource(
```

Comment:
let's add a comment describing why we need this

Reply:
From the command (and from the code) I can understand we're allowing inbound SSH connection but it's not clear "why" we need to open SSH from everywhere. I think it's because the test instance needs to connect to it to execute commands in the node, but we should clarify this in the comment.